kernel/mutex.c

```diff
@@ -194,6 +201,17 @@ int z_impl_k_mutex_lock(struct k_mutex *mutex, k_timeout_t timeout)
 	SYS_PORT_TRACING_OBJ_FUNC_EXIT(k_mutex, lock, mutex, timeout, -EAGAIN);
 
 	return -EAGAIN;
+#else
+	ARG_UNUSED(mutex);
+	ARG_UNUSED(timeout);
+
+	__ASSERT(!arch_is_in_isr(), "mutexes cannot be used inside ISRs");
+
+	SYS_PORT_TRACING_OBJ_FUNC_ENTER(k_mutex, lock, mutex, timeout);
+	SYS_PORT_TRACING_OBJ_FUNC_EXIT(k_mutex, lock, mutex, timeout, 0);
+
+	return 0;
+#endif /* CONFIG_MULTITHREADING */
```

Comment:
Some parts are duplicated in both sections of the`#ifdef`. They can probably just be refactored to the top of the function, outside the `#ifdef`, and then maybe the `#else` can be removed.

Reply:
See fixup commit. In this case the return value differ and the `#else` seems to be necessary. If you see a better way to express this I'm happy to adjust.